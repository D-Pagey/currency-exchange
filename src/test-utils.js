import React from 'react';
import { render } from '@testing-library/react';
import { AccountsContext, initialAccounts } from './components/ProviderAccounts';

const customRender = (ui, accountsContextValue = {}) => {
    return render(
        <AccountsContext.Provider
            value={{
                accounts: initialAccounts,
                setAccounts: () => null,
                ...accountsContextValue,
            }}
        >
            {ui}
        </AccountsContext.Provider>,
    );
};

export * from '@testing-library/react';

export { customRender as render };
