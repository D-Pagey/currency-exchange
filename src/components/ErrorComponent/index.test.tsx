import React from 'react';
import { render } from '../../test-utils';
import { ErrorComponent } from '.';

const props = {
    message: 'API limit reached',
};

describe('ErrorComponent', () => {
    it('should render', () => {
        const { container } = render(<ErrorComponent {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
