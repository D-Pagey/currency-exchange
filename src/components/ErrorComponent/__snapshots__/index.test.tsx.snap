// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent should render 1`] = `
<div
  class="sc-AxjAm kyFXYz"
  data-testid="errorComponent"
>
  <img
    alt="alert"
    class="sc-AxirZ ienVnP"
    src="alert.svg"
  />
  <p
    class="sc-AxiKw dQKCBV"
  >
    API limit reached
  </p>
</div>
`;
